### PR TITLE
J3RN's solutions

### PR DESCRIPTION
**Pull request by J3RN**
I'd love feedback! Also enjoy the approximately correct growth and apple harvest numbers!

### CODE REVIEW

File: lib/tree.rb [hunk 1]
```diff
@@ -1,38 +1,58 @@
...
   end
 end
 
-class Apple <
-  attr_reader #what should go here 
+class Apple < Fruit
+  attr_reader :color, :diameter
 
   def initialize(color, diameter)
+    super()
```
Comment (jaybobo) [inline lib/tree.rb:52]: Nice use of inheritance here

File: lib/tree.rb [hunk 2]
```diff
@@ -1,38 +1,58 @@
...
   end
 
   def pick_an_apple!
-    raise NoApplesError, "This tree has no oranges" unless self.any_apples?
+    raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.pop
   end
 
   def dead?
+    not @alive
```
Comment (jaybobo) [inline lib/tree.rb:36]: besides being more readable is there a difference between `not @alive` and `!@alive`?

File: spec/tree_spec.rb [hunk 3]
```diff
@@ -2,13 +2,76 @@
...
 
 describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+  before(:each) do
+    @tree = AppleTree.new
+  end
+
+  it 'should age' do
+    before_age = @tree.age
```
Comment (jaybobo) [inline spec/tree_spec.rb]: Look into `let` and its advantages.
https://www.relishapp.com/rspec/rspec-core/v/2-5/docs/helper-methods/let-and-let

File: lib/tree.rb (hunk 2 above)
Comment (J3RN) [inline lib/tree.rb:36]: There's a difference with regards to precedence and implementation. However, in this instance it was a readability decision. That said, I'm not tied to `not`, and could be persuaded to use `!` instead.

File: spec/tree_spec.rb (hunk 3 above)
Comment (J3RN) [inline spec/tree_spec.rb]: Good call! Switched!

**Pull request closed without merging**